app/routes/datasets/__init__.py

```diff
@@ -82,3 +97,63 @@ async def validate_creation_options(
         await validator[input_data["asset_type"]](dataset, version, input_data)
     except KeyError:
         pass
+
+
+# I cannot seem to satisfy mypy WRT the type of this default dict. Last thing I tried:
+# DefaultDict[str, Callable[[str, str, int, int, ...], List[str]]]
```

Comment:
I might wait on this for a different PR and consult with @dmannarino , this was just a copy and paste from a different file because I wanted to re-use this existing function in a different class.